daemon/info.go

```diff
@@ -233,23 +239,48 @@ func memInfo() *system.MemInfo {
 	return memInfo
 }
 
+// this caches the result of operatingSystem below, for efficiency
+var operatingSystemResult string
+
 func operatingSystem() string {
-	var operatingSystem string
-	if s, err := operatingsystem.GetOperatingSystem(); err != nil {
-		logrus.Warnf("Could not get operating system name: %v", err)
-	} else {
-		operatingSystem = s
+	defer metrics.StartTimer(hostInfoFunctions.WithValues("operating_system"))()
+
+	if operatingSystemResult == "" {
+		var operatingSystem string
```

Comment:
nit; wondering if we should just move this variable out instead of introducing the `operatingSystemResult` variable (definitely open to suggestions there)